### ISSUE TITLE
Unexpected error position

### ISSUE THREAD

**Issue by t-ce** [eslint/eslint#8053]
I'm using Atom with eslint@3.15.0 , npm 3.10.10 , note v6.9.4

My .eslintrc.js is
```js
module.exports = {
  root: true,
  parser: 'babel-eslint',
  parserOptions: { sourceType: 'module' },
  extends: 'standard',
  plugins: [ ],
  'rules': {
    'arrow-parens': 0,            // allow paren-less arrow functions
    'generator-star-spacing': 0,  // allow async-await
    'no-debugger': 0              // allow debugger during development
  }
}
```

## Problem

**What happen:**
1) This code gives no error:
```js
// String interpolation
var user = {username: 'Alice', hashedPassword: '73895'}
console.log(`Username: ${user.username}, password: ${user.hashedPassword}`)
```
2) This code gives two errors:
```js
// String interpolation
var user = {username: 'Alice', hashedPassword: '73895'}
console.log(`Username: ${user.username}, password: ${user.hashedPassword}`) // Line 3
// Immediately Invoked Function Expressions (IIFEs)
(function f6 () { console.log('IIFE') })()                                  // Line 5
```
At line 3 I get `func-call-spacing` and at line 5 I get `no-unexpected-multiline`.

**What I expect**
I have no problem that at line 5 there is an error (I still have to discover how to write an IIFE with ESLint) but at line 3 no error should be signaled.

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, I think this is actually working as intended -- due to the way that semicolons work in JavaScript,  you're *calling* the result of `console.log()` on line 3, and your function on line 5 is getting parsed as an argument. So your code is equivalent to

```js
console.log('...')(function f6 () { /* ... */ })()
```

...which is probably not what you were trying to do. The `no-unexpected-multiline` rule is reporting an error to warn you about this.

To fix the issue, you can put a semicolon somewhere between line 3 and line 5:

```js
console.log('...');

;(function f6 () { /* ... */ })()
```

**Comment by t-ce** [comment]
thank you! incredible. elsewhere I get a `semi` error (extra semicolon) and there I need to put one.
That is, in the `semi` rule you (ESLint project) have extra programmed an exception to accept the semicolon when a IIFE follows?
This issue can be closed, thank you.

**Comment by not-an-aardvark** [comment]
The `semi` rule only warns if you're missing a semicolon at the end of a statement, but since the IIFE gets interpreted as a function call, the end of your `console.log` line isn't actually the end of a statement.
